backends/imgui_impl_wgpu.cpp

```diff
@@ -455,7 +455,10 @@ static void ImGui_ImplWGPU_CreateFontsTexture()
         tex_desc.dimension = WGPUTextureDimension_2D;
         tex_desc.size.width = width;
         tex_desc.size.height = height;
+        tex_desc.size.depthOrArrayLayers = 1;
+#if !defined(__EMSCRIPTEN__)
```

Comment:
nit: Dawn in native should understand `depthOrArrayLayers` too so this ifdef might not be necessary.

Reply:
This is actually a work-around for an issue I've encountered in Dawn (latest commit 30.4.). The validator checks for `depth != 0`, which triggers an error.

Reply:
It shouldn't AFAIK: https://source.chromium.org/chromium/chromium/src/+/main:third_party/dawn/src/dawn_native/Texture.cpp;drc=2dd2d67dbcbbf0a241fd13d4fccbc049ee39bfd4;l=288

What error message do you get?

Reply:
Thanks for pointing it out. I had another look at the code. It seems like dawn is trying to fix 'wrong' settings in the wrong way:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/dawn/src/dawn_native/Texture.cpp;drc=2dd2d67dbcbbf0a241fd13d4fccbc049ee39bfd4;l=251

Reply:
Ah, that's because it is assuming there is the C++ structure defaulting happening. https://source.chromium.org/chromium/chromium/src/+/main:out/Debug/gen/third_party/dawn/src/include/dawn/webgpu_cpp.h;drc=d3d0337463160599bde3ef97db67071148652b9c;l=1203

Keeping as is sounds good, but soon (as in this week) Dawn will remove depth since we're past the deprecation period.

Reply:
Well, the default initialization in the cpp header explains a lot. I am going to convert the implementation eventually to use the cpp version.

I am going to remove `depth` in this case, as Dawn users will probably stay at head anyways (this implementation anyway already uses most of the latest changes().